### PR TITLE
[tomlplusplus] Update to 2.5.0

### PR DESCRIPTION
**Pull request by qudix**
**Describe the pull request**

- #### What does your PR fix?  
  Updates tomlplusplus to 2.5.0

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  Just an update

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes

### CODE REVIEW

File: ports/tomlplusplus/vcpkg.json [hunk 1]
```diff
@@ -1,7 +1,6 @@
 {
   "name": "tomlplusplus",
-  "version-string": "2.4.0",
-  "port-version": 1,
+  "version-string": "2.5.0",
```
Comment (PhoebeHui) [inline ports/tomlplusplus/vcpkg.json:3]:
```suggestion
  "version": "2.5.0",
```

**Review by PhoebeHui — CHANGES_REQUESTED** [review]
@Qudix, thanks for the PR!

Could you address the review suggestion?

**Comment by Qudix** [comment]
I'm not sure what's wrong with CI, seems unrelated?

**Comment by PhoebeHui** [comment]
The CI failed with the following failure, It looks relate to https://github.com/microsoft/vcpkg/pull/20140

@BillyONeal, do you know how to find the 2 unexpected failures port?

eg: x86-windows:
```
Expected 28 cascaded failures, but there were 31 cascaded failures.
##[error]vcpkg ci failed
At C:\a\1\s\scripts\azure-pipelines\test-modified-ports.ps1:174 char:5
+     throw "vcpkg ci failed"
+     ~~~~~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (vcpkg ci failed:String) [], RuntimeException
+ FullyQualifiedErrorId : vcpkg ci failed
##[error]PowerShell exited with code '1'.
```

**Comment by PhoebeHui** [comment]
```
x86-windows:                   Expected 28 cascaded failures, but there were 31 cascaded failures.
x64-windows:                   Expected 21 cascaded failures, but there were 24 cascaded failures.
x64-windows-static:            Expected 59 cascaded failures, but there were 62 cascaded failures.
x64-windows-static-md:         Expected 53 cascaded failures, but there were 56 cascaded failures.
x64-uwp:                       Expected 345 cascaded failures, but there were 342 cascaded failures.
arm64-windows:                 Expected 228 cascaded failures, but there were 226 cascaded failures.
arm-uwp:                       Expected 345 cascaded failures, but there were 343 cascaded failures.
x64-osx:                       Pass
x64-linux:                     Pass
```

**Comment by BillyONeal** [comment]
>@BillyONeal, do you know how to find the 2 unexpected failures port?

It was just a count lock, not a per-port lock. We backed out the check entirely.

**Review by PhoebeHui — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your update!

**Pull request merged**